code/framework/encryption/src/main/java/io/cattle/platform/framework/encryption/impl/Aes128Encrypter.java

```diff
@@ -0,0 +1,72 @@
+package io.cattle.platform.framework.encryption.impl;
+
+import io.cattle.platform.framework.encryption.Encrypter;
+import io.cattle.platform.framework.encryption.EncryptionUtils;
+
+import java.security.Key;
+import java.security.SecureRandom;
+import javax.crypto.Cipher;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+
+import org.apache.commons.codec.binary.Hex;
+
+public class Aes128Encrypter extends Encrypter {
+    private String encryptionKey;
+
+    private Cipher encrypter;
+    private Cipher decrypter;
+    private static final String NAME = "AES128";
+    private Key key;
+    private SecureRandom rn;
+
+    public Aes128Encrypter() throws Exception {
+        rn = new SecureRandom();
+
+        encrypter = Cipher.getInstance("AES/CBC/PKCS5PADDING", "SunJCE");
+        try {
+            encryptionKey = EncryptionUtils.getKeyFromFile(NAME + "Key");
+        } catch (NullPointerException e) {
+            byte[] encryptionKeyBytes = new byte[8];
+            rn.nextBytes(encryptionKeyBytes);
+            encryptionKey = String.valueOf(Hex.encodeHex(encryptionKeyBytes));
+            EncryptionUtils.saveKeyToFile(NAME + "Key", encryptionKey);
+        }
+
+        key = new SecretKeySpec(encryptionKey.getBytes("UTF-8"), "AES");
+        decrypter = Cipher.getInstance("AES/CBC/PKCS5PADDING", "SunJCE");
+    }
+
+
+    public String encrypt(String plainText) {
+        try {
+            byte[] ivbytes = new byte[8];
+            rn.nextBytes(ivbytes);
+            String IV = String.valueOf(Hex.encodeHex(ivbytes));
+            encrypter.init(Cipher.ENCRYPT_MODE, key, new IvParameterSpec(IV.getBytes("UTF-8")));
+            return IV + ":" + String.valueOf(Hex.encodeHex(encrypter.doFinal(plainText.getBytes("UTF-8"))));
+        } catch (Exception e){
+            e.printStackTrace();
+            throw new IllegalArgumentException(e);
+        }
+    }
+
+    public String decrypt(String value) {
+        try {
+            String[] split = value.split(":", 2);
+            decrypter.init(Cipher.DECRYPT_MODE, key, new IvParameterSpec(split[0].getBytes("UTF-8")));
+            return new String(decrypter.doFinal(Hex.decodeHex(split[1].toCharArray())), "UTF-8");
+        } catch (Exception e){
+            throw new IllegalArgumentException();
```

Comment:
Can we provide more context in this exception? Some message indicating on what exactly failed